lib/_http_outgoing.js

```diff
@@ -602,6 +604,24 @@ OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
 };
 
 
+// Returns an array of the names of the current outgoing raw headers.
+OutgoingMessage.prototype.getRawHeaderNames = function getRawHeaderNames() {
+  const headersMap = this[kOutHeaders];
+  const headers = Array(ObjectKeys(headersMap).length);
+
+  if (headersMap !== null) {
+    const values = ObjectValues(headersMap);
```

Comment:
Can we avoid calling `%Object.keys%` here? We can get the same info from `%Object.values%`, right? Also, if `headersMap` is `null`, wouldn't `%Object.keys%` throw?
```suggestion
  if (headerMap == null) return [];
  
  const values = ObjectValues(headersMap);
  const headers = Array(values.length);

  {
```

Reply:
Good call, pushed :+1:

Reply:
I missed the `length` comment, but I can add that if needed?

Reply:
Btw `values.length` is stored in `l` so that's only one assignment + accessing the property added.

Reply:
I think we can either:

1. store the value in a variable and read the variable to avoid looking up the property value several time.
2. access the property each time and rely on the compiler optimisations.

The current implementation access the property twice before putting its value in a variable, that seems unnecessary.

Reply:
> The current implementation access the property twice before putting its value in a variable, that seems unnecessary.

Exactly. Not sure if one additional property access call on top changes anything, but I can update the code if you want. Your call.

Reply:
Yes, I'd prefer to see either of the approach described in my comment above.